Source/CarthageKit/Project.swift

```diff
@@ -1042,26 +1042,73 @@ public final class Project { // swiftlint:disable:this type_body_length
 				let derivedDataVersioned = derivedDataPerDependency.appendingPathComponent(version.commitish, isDirectory: true)
 				options.derivedDataPath = derivedDataVersioned.resolvingSymlinksInPath().path
 
-				return build(dependency: dependency, version: version, self.directoryURL, withOptions: options, sdkFilter: sdkFilter)
-					.map { producer -> BuildSchemeProducer in
-						return producer.flatMapError { error in
-							switch error {
-							case .noSharedFrameworkSchemes:
-								// Log that building the dependency is being skipped,
-								// not to error out with `.noSharedFrameworkSchemes`
-								// to continue building other dependencies.
-								self._projectEventsObserver.send(value: .skippedBuilding(dependency, error.description))
-								return .empty
+				return self
+					.dependencySet(for: dependency, version: version)
+					.flatMap(.concat) { dependencies -> SignalProducer<(), CarthageError> in
+						// Don't create symlink the build folder if the dependencies doesn't have
+						// any Carthage dependencies.
+						if dependencies.isEmpty {
+							return .empty
+						}
+						return symlinkBuildPath(for: dependency, rootDirectoryURL: self.directoryURL)
+					}
```

Comment:
Here's the actual fix. We use similar logic for not creating `Carthage/Checkouts/`.

This call to `symlinkBuildPath` used to be inside the `build` call right after this.